plugins/faucet/plugin.go

```diff
@@ -38,26 +35,20 @@ const (
 
 var (
 	// Plugin is the "plugin" instance of the faucet application.
-	Plugin                   *node.Plugin
-	_faucet                  *StateManager
-	powVerifier              = pow.New()
-	fundingWorkerPool        *workerpool.NonBlockingQueuedWorkerPool
-	fundingWorkerCount       = runtime.GOMAXPROCS(0)
-	fundingWorkerQueueSize   = 500
-	preparingWorkerPool      *workerpool.NonBlockingQueuedWorkerPool
-	preparingWorkerCount     = runtime.GOMAXPROCS(0)
-	preparingWorkerQueueSize = MaxFaucetOutputsCount + 1
-	targetPoWDifficulty      int
-	// blacklist makes sure that an address might only request tokens once.
-	blacklist         *orderedmap.OrderedMap[string, bool]
-	blacklistCapacity int
-	blackListMutex    sync.RWMutex
+	Plugin              *node.Plugin
+	_faucet             *Faucet
+	powVerifier         = pow.New()
+	requestChanSize     = 100
+	requestChan         = make(chan *faucet.Payload, requestChanSize)
+	targetPoWDifficulty int
+
 	// signals that the faucet has initialized itself and can start funding requests.
 	initDone     atomic.Bool
 	bootstrapped chan bool
 
-	waitForManaWindow = 5 * time.Second
-	deps              = new(dependencies)
+	waitForManaWindow   = 5 * time.Second
+	faucetRequestWindow = 3 * time.Minute
```

Comment:
After all we might not need this because the faucet does not pledge tokens when it's not bootstrapped, which ultimately covers this check.